src/state/selectors/axisSelectors.ts

```diff
@@ -870,12 +870,20 @@ function getD3ScaleFromType(realScaleType: string | undefined) {
 }
 
 function combineScaleFunction(
+  axis: BaseAxis,
   realScaleType: string | undefined,
   axisDomain: NumberDomain | CategoricalDomain,
   axisRange: [number, number],
 ): RechartsScale | undefined {
+  if (axisDomain == null) {
+    return undefined;
+  }
+  if (typeof axis.scale === 'function') {
+    // @ts-expect-error we're going to assume here that if axis.scale is a function then it is a d3Scale function
+    return axis.scale.copy().domain(axisDomain).range(axisRange);
```

Comment:
I guess we'll find out when we release if this is an issue 😅

Reply:
This has always been the case so it shouldn't be an issue.

Reply:
Except before my changes this assumption was hidden behind an ocean of `any`s 😬

Reply:
Of course it was haha